### PR TITLE
Adds offline SLAM; build maps as fast as possible.

### PR DESCRIPTION
**Pull request by damonkohler**

### CODE REVIEW

File: cartographer_ros/cartographer_ros/offline_node_main.cc [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    urdf_filename, "",
+    "URDF file that contains static links for your sensor configuration.");
+
+namespace cartographer_ros {
+namespace {
+
+constexpr int kLatestOnlyPublisherQueueSize = 1;
+
+// Default topic names; expected to be remapped as needed.
+constexpr char kLaserScanTopic[] = "scan";
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/offline_node_main.cc]: Add a `TODO` that this should be deduplicted in a follow up PR?

File: cartographer_ros/cartographer_ros/offline_node_main.cc [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+constexpr char kFinishTrajectoryServiceName[] = "finish_trajectory";
+constexpr char kClockTopic[] = "clock";
+
+void Run() {
+  auto file_resolver = cartographer::common::make_unique<
+      cartographer::common::ConfigurationFileResolver>(
+      std::vector<string>{FLAGS_configuration_directory});
+  const string code =
+      file_resolver->GetFileContentOrDie(FLAGS_configuration_basename);
+  cartographer::common::LuaParameterDictionary lua_parameter_dictionary(
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/offline_node_main.cc]: Can this be extracted to some common function?

File: cartographer_ros/cartographer_ros/offline_node_main.cc [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+namespace cartographer_ros {
+namespace {
+
+constexpr int kLatestOnlyPublisherQueueSize = 1;
+
+// Default topic names; expected to be remapped as needed.
+constexpr char kLaserScanTopic[] = "scan";
+constexpr char kMultiEchoLaserScanTopic[] = "echoes";
+constexpr char kPointCloud2Topic[] = "points2";
+constexpr char kImuTopic[] = "imu";
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/offline_node_main.cc]: Should we make these already resolved names to make sure that all topics contained in the bag can be remapped without interfering with our default topic names? I.e. Do not call resolve on the `k...Topic` names?

File: cartographer_ros/cartographer_ros/offline_node_main.cc (hunk 1 above)
Comment (damonkohler) [inline cartographer_ros/cartographer_ros/offline_node_main.cc]: Done.

File: cartographer_ros/cartographer_ros/offline_node_main.cc (hunk 2 above)
Comment (damonkohler) [inline cartographer_ros/cartographer_ros/offline_node_main.cc]: Added a todo.

**Pull request merged**